base/compiler/abstractinterpretation.jl

```diff
@@ -544,6 +544,10 @@ function abstract_call_method(interp::AbstractInterpreter,
     sigtuple = unwrap_unionall(sig)
     sigtuple isa DataType || return MethodCallResult(Any, false, false, nothing, Effects())
 
+    if is_noinfer(method)
```

Comment:
We should make this controllable by the absint. GPUCompiler may choose to ignore this and force full inference.

cc: @maleadt

Reply:
I'm fine with implementing a configuration here but maybe we can delay it until it turns out that GPUCompiler (or other external compilation pipeline) really needs it? Since `@nospecializeinfer` does not prohibit constant prop' and such, I'm still wondering if it can cause unwanted inference regression.